containers/podman-desktop
# chore(tests): add podman desktop update e2e test

### What does this PR do?
Brings new E2E update test. The test is intended to be run solely in the specific test environment. Windows (now) and Mac will be target env. 

The update e2e test will run after each merge, and in the PR check if it was initialized by dependabot where we saw the most problems.

### Screenshot / video of UI



### What issues does this PR fix or reference?
#7931 


### How to test this PR?
On windows only now. See the details in the linked issue. You need change version of PD in package.json to something like `1.0.0` and then run `yarn test:e2e:update`. Also, if you want to perform actual update during the testing, add env. var: `UPDATE_PODMAN_DESKTOP=true`
In order to be able to run the update in locally built podman desktop, you need to set `ELECTRON_ENABLE_INSPECT` env. var. and build the bits to have correct artifacts: `pnpm compile:current --win nsis`. It will produce executable podman desktop binary, which will offer an update option. In order to execute the e2e tests now with built executable, use `PODMAN_DESKTOP_BINARY="C:\path\to\dist\win-unpacked\Podman Desktop.exe"` when running test target.
